content/chapters/software-stack/lecture/slides/software-stack.md

```diff
@@ -0,0 +1,91 @@
+---
+
+## Software Stack
+
+---
+
+### Flexibility of Software
+
+TODO: Diagram with users, hardware and software
+
+----
+
+### Flexibility of Software
+
+* Hardware provides basic actions
+* Users request rich features
+* Software bridges the gap between user demands and hardware possibility
+
+----
+
+### Hardware vs Software
+
+| Hardware                         | Software                                   |
+| :------------------------------: | :----------------------------------------: |
+| TODO                             | TODO                                       |
+| TODO                             | TODO                                       |
+| TODO                             | TODO                                       |
+
+---
+
+### Properties of Software
+
+* Reusability
+* Portability
+* Configurability / Flexibility
+
+----
+
+### Reusability
+
+TODO: diagram
+
+----
+
+### Portability
+
+TODO: diagram
+
+----
+
+### Configurability
+
+TODO: diagram
+
+---
+
+### Interface and Implementation
+
+TODO: diagram
+
+----
+
+### Portability vs Performance
+
+TODO
+
+---
+
+### Software Stack
+
+TODO: diagram
+
+----
+
+### Software Stack
+
+* Each layer provides an interface to the upper layer
+* Richer set of features are implemented by the upper layers
+* Similar to the networking stack
+
+----
+
+### Android Software Stack
+
+TODO: image
+
+----
+
+### iOS Software Stack
+
+TODO: image
```

Comment:
These chapters are also present in `modern-software-stacks.md`. Is it intended or by mistake?

Reply:
It's not a mistake. I'm fond of doing recaps on given concepts. It sticks.